app/src/main/java/com/duckduckgo/app/trackerdetection/UrlToTypeMapper.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.trackerdetection
+
+import android.webkit.MimeTypeMap
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+interface UrlToTypeMapper {
+    fun map(
+        url: String,
+        requestHeaders: Map<String, String>,
+    ): String?
+}
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
```

Comment:
I don't think we need a single instance here because there is no state in this class.